include/query.php

```diff
@@ -174,6 +174,14 @@ public function filter_query( $lang ) {
 				}
 			}
 		} else {
+			// Set the language correctly in the query, since WordPress merges the language with the other query vars when the operator is OR.
+			if ( isset( $this->query->tax_query->queried_terms['language'] ) && 1 === count( $this->query->tax_query->queried_terms['language']['terms'] ) ) {
```

Comment:
We will have the same problem when assigning multiple language.
I propose to modify `set_language()` for it to accept `PLL_Language|PLL_Language[]`.